src/test/scala/chiselTests/properties/PropertySpec.scala

```diff
@@ -585,4 +587,74 @@ class PropertySpec extends ChiselFlatSpec with MatchesAndOmits {
     ))
     e.getMessage should include("Properties do not support hardware printing: 'in_bar', in module 'PropertySpec_Anon'")
   }
+
+  it should "not be able to get the ClassType of a Property[T] if T != ClassType" in {
+    assertTypeError("""
+      val intProp = Property[Int](123)
+      val foobar = Property[intProp.ClassType]()
+    """)
+  }
+
+  it should "emit correct types for all the ways of creating class references and properties" in {
+    val chirrtl = ChiselStage.emitCHIRRTL(new RawModule {
+      val cls = ClassType("MyClass")
```

Comment:
I dont think we want this to work, `Class.unsafeGetDynamicObject` is supposed to be the way to reflectively talk about classes. Now that returns a DynamicObject so I get you need a way to talk about Classes themselves, although perhaps the type should be a method on the Object rather than a `ClassType`? Eventually we'll make D/I work for Classes and then that will be the way you're typically supposed to create Classes.

At the very least can it have an annoying method name (rather than "apply") to indicate it's unsafe?

Reply:
make `ClassType.apply` private and added `ClassType.unsafeGetClassTypeByName`